Fixes part of #1096: Crashlytics implementation in domain layer

Sarthak2601 opened the pull request:
<!--
  - Thanks for submitting code to Oppia! Please fill out the following as part of
  - your pull request so we can review your code more easily.
  -->

## Explanation
<!--
  - Explain what your PR does. If this PR fixes an existing bug, please include
  - "Fixes #bugnum:" in the explanation so that GitHub can auto-close the issue
  - when this PR is merged.
  -->
Fixes part of #1096 
- Implemented Crashlytics in domain layer.
- Added tests as well. 
## Checklist
<!-- Please tick the relevant boxes by putting an "x" in them. -->
- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [ ] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

Sarthak2601 committed:
crashlytics + tests : domain

vinitamurthi commented on domain/src/test/java/org/oppia/domain/audio/AudioPlayerControllerTest.kt:
```diff
@@ -27,6 +27,7 @@ import org.mockito.Mockito.verify
+import org.oppia.testing.TestLogReportingModule
 import org.oppia.util.caching.CacheAssetsLocally
 import org.oppia.util.data.AsyncResult
 import org.oppia.util.logging.EnableConsoleLog
```
Given that we are now creating our own test cases per file instead of adding to other tests, you should add tests in each of these files for a failure case and check if the crash was logged. (Here and below)

Sarthak2601 replied:
These files only have dependency over the module but don't have any failure cases that actually call the crashLogger.

vinitamurthi replied:
This file is calling the crashLogger. it may not have a failure test case in the file but you need to add one since it is calling the crash logger

Sarthak2601 replied:
Done.

vinitamurthi commented on domain/src/test/java/org/oppia/domain/classify/AnswerClassificationControllerTest.kt:
```diff
@@ -485,7 +486,8 @@ class AnswerClassificationControllerTest {
   @Component(modules = [
     TestModule::class, ContinueModule::class, FractionInputModule::class, ItemSelectionInputModule::class,
```
Add a test for one failure case

Sarthak2601 replied:
They only have dependency, no failure cases.

vinitamurthi replied:
No , these files have a logException() command present in the file (See the first two files in this PR)

Sarthak2601 replied:
Done

vinitamurthi replied:
You can revert the lint changes in this file

vinitamurthi reviewed (changes requested):
The AudioPlayerController and AnswerClassificationController both are logging crashes, so we need to add test cases for them

Sarthak2601 committed:
Test in AudioPlayerControllerTest | Removed logException from AnswerClassificationController

Sarthak2601 commented:
> The AudioPlayerController and AnswerClassificationController both are logging crashes, so we need to add test cases for them

Added test case in `AudioPlayerController` and removed crash logger implementation from `AnswerClassificationController` as it throws fatal error which will be caught by Crashlytics anyway.

vinitamurthi commented on domain/src/main/java/org/oppia/domain/classify/AnswerClassificationController.kt:
```diff
@@ -1,10 +1,10 @@
 package org.oppia.domain.classify
 
```
I think you can revert all the changes in this file. It looks like this is just a linting change and that will be handled later

Sarthak2601 commented:
Closing because of implementational changes.
